src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -200,6 +200,43 @@ ShardLength(uint64 shardId)
 
 
 /*
+ * NodeHasShardPlacements returns whether any shards are placed on this node
+ */
+bool
+NodeHasShardPlacements(char *nodeName, int32 nodePort)
```

Comment:
pushed but I'm not sure this is a good idea. You'll have some unintuitive behavior and no warning when you mark those rows as active again. We should probably also throw an error when you try to modify those rows, saying the node doesn't exist and they can't be activated until it does.